### DISCUSSION TITLE
Teensy 4.1 with ST7789 240x240 display

### DISCUSSION THREAD

**Discussion by jlt206** [Bodmer/TFT_eSPI#2735] [Compatible support libraries]
I am changing my board to a Teensy 4.1 as it can support 2 SPI displays, but can't find a set-up file for this board.

**Comment by jlt206** [comment]
[Teensy_PimoRound.txt](https://github.com/Bodmer/TFT_eSPI/files/11995006/Teensy_PimoRound.txt)
Teensy_PimoRound.h created and working.

**Reply by jlt206** [reply]
I have been able to get this working with one Display but not 2 so far. Becuase of the Display not having a RESET, I am not able to daisy chain them to use the Dual Display Set-up.

However, the Teensy 4.1 has more than one SPI port. So I have been trying to use SPI0 and SPI1 but need some gudiance to see if this is possible with your libraries?

At the moment I am trying to figure out how to use both SPI interfaces to drive the screens seperately. But have been successful with making both work indipendalty as yet. I can still only get one working on SPI0.

![PXL_20230710_090123147 MP](https://github.com/Bodmer/TFT_eSPI/assets/4954838/2ac8583d-193f-4b6b-8cda-d23aa7e46da9)
![T4 1-Cardlike](https://github.com/Bodmer/TFT_eSPI/assets/4954838/8f678569-d5ae-4f3f-8c2e-89e74f6dfde2)

**Comment by jlt206** [comment]
With some help from the Teensy Forum, I was able to get both screens working on seperate SPI buses.
Not using TFT_eSPI.h yet as I'm not sure this is provided for?

#include "SPI.h"
#include "Wire.h"
#include <Adafruit_ST7789.h> // Hardware-specific library for ST7789

#define TFT_CS_1 10
#define TFT_DC_1 2

#define TFT_CS_2 0
#define TFT_DC_2 9

#define TFT_RST 32                                            

Adafruit_ST7789 tft1 = Adafruit_ST7789(TFT_CS_1, TFT_DC_1, TFT_RST);
Adafruit_ST7789 tft2 = Adafruit_ST7789(&SPI1, TFT_CS_2, TFT_DC_2, TFT_RST);

void setup(){
  Serial.begin(115200);
//  while (!Serial);
  tft1.init(240, 240);
  tft1.setRotation(-2);
  tft1.fillScreen(0x0000);
  tft1.setCursor(60, 120);
  tft1.setTextColor(0xFFFF);
  tft1.setTextSize(2);
  tft1.println("Display 1");
  delay(2000);
  tft1.fillScreen(0x0000);

  pinMode(9, OUTPUT);
  SPI1.begin();
  
  tft2.init(240, 240);
  tft2.setRotation(0);
  tft2.fillScreen(0x0000);
  tft2.setCursor(60, 120);
  tft2.setTextColor(0xFFFF);
  tft2.setTextSize(2);
  tft2.println("Display 2");
  delay(2000);
  tft1.fillScreen(0x0000);
}

void loop()
{
  tft1.setCursor(60, 120);
  tft1.println("Display 1");

  tft2.setCursor(60, 120);
  tft2.println("Display 2");

  delay(1000);
}
![PXL_20230710_133627597](https://github.com/Bodmer/TFT_eSPI/assets/4954838/0f842a35-4302-49a3-978e-df0683701369)

**Comment by Bodmer** [comment]
Two displays can be connected to the same SPI bus, but the display chip selects must be managed by the sketch. The are some YouTube videos that show how this is done.

**Comment by jlt206** [comment]
HI Bodmer, Thanks for the message.

I have solved the problem with a Teensy 4.1 which allows me to use 2 seperate SPI buses.

But I haven't figured out to use this configuration with your libraries.

![PXL_20230711_081543953](https://github.com/Bodmer/TFT_eSPI/assets/4954838/6f669685-0f70-497c-9ae4-f3de126a4e73)
